samples/load_data_by_post.py

```diff
@@ -0,0 +1,82 @@
+import json
+import httplib2
+from samples.utils import poll_job
+from oauth2client.client import GoogleCredentials
+from googleapiclient.discovery import build
+
+
+# [START make_post]
+def make_post(http, schema, data, projectId, datasetId, tableId):
+    url = ('https://www.googleapis.com/upload/bigquery/v2/projects/' +
+           projectId + '/jobs')
+    # Create the body of the request, separated by a boundary of xxx
+    resource = ('--xxx\n' +
+                'Content-Type: application/json; charset=UTF-8\n' + '\n' +
+                '{\n' +
+                '   "configuration": {\n' +
+                '     "load": {\n' +
+                '       "schema": {\n'
+                '         "fields": ' + str(schema) + '\n' +
+                '      },\n' +
+                '      "destinationTable": {\n' +
+                '        "projectId": "' + projectId + '",\n' +
+                '        "datasetId": "' + datasetId + '",\n' +
+                '        "tableId": "' + tableId + '"\n' +
+                '      }\n' +
+                '    }\n' +
+                '  }\n' +
+                '}\n' +
+                '--xxx\n' +
+                'Content-Type: application/octet-stream\n' +
+                '\n')
+    # Append data to the request body
+    resource += data
+
+    # Signify the end of the body
+    resource += ('--xxx--\n')
+
+    headers = {'Content-Type': 'multipart/related; boundary=xxx'}
+
+    return http.request(url,
+                        method='POST',
+                        body=resource,
+                        headers=headers)
+    # [END make_post]
+
+
+# [START main]
+def main():
+    credentials = GoogleCredentials.get_application_default()
+    http = credentials.authorize(httplib2.Http())
+    projectId = raw_input('Choose your project ID: ')
+    datasetId = raw_input('Choose a dataset ID: ')
+    tableId = raw_input('Choose a table name to load the data to: ')
+    schema_path = raw_input(
+            'Enter the path to the schema file for the table: ')
+
+    with open(schema_path, 'r') as schema_file:
+        schema = schema_file.read()
+
+    data_path = raw_input('Enter the path to your data file: ')
+
+    with open(data_path, 'r') as data_file:
+        data = data_file.read()
+
+    resp, content = make_post(http,
+                              schema,
+                              data,
+                              projectId,
+                              datasetId,
+                              tableId)
+
+    if resp.status == 200:
+        job_resource = json.loads(content)
+        service = build('bigquery', 'v2', credentials=credentials)
```

Comment:
Oh I see. Does setting REMOTE_ADDR help? I assume you can benefit from http cache in that case. I think this is enough.

My point is that if you need to cache the discovery doc, it will partially kill the handiness of our discovery based API.

What do you think?